app/javascript/controllers/image_upload_controller.js

```diff
@@ -1,56 +1,80 @@
 import { Controller } from "@hotwired/stimulus"
 
 export default class extends Controller {
-  static targets = ["file", "content"]
+  static targets = [ "file", "content", "preview" ]
 
   connect() {
-    this.dropHandler()
-    this.pasteHandler()
+    this.fileTarget.addEventListener("change", this.boundPreviewUpdate)
+    this.contentTarget.addEventListener("dragover", this.boundIgnore)
+    this.contentTarget.addEventListener("dragenter", this.boundIgnore)
```

Comment:
I'm surprised that we even need to listen for these events if they're just going to be "prevented" -- but I really have very little experience with the drag and drop API. What happens when these aren't blocked?

Reply:
Good call. I didn't actually try it without them. I just did and only one is needed for one case so I've removed it and refactored the code with the single suppression that's needed.